app/src/test/java/com/example/trainingintive/viewmodels/DogImageViewModelTest.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.example.trainingintive.TestSchedulersProvider
+import com.example.trainingintive.model.DogImageUrl
+import com.example.trainingintive.repository.DogImageRepository
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Single
+import org.amshove.kluent.`should be equal to`
+import org.junit.Rule
+import org.junit.Test
+
+class DogImageViewModelTest {
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    val sampleImageUrl = "https:\\/\\/images.dog.ceo\\/breeds\\/bulldog-french\\/n02108915_9666.jpg"
+    val sampleStatus = "success"
+    val sampleDog = DogImageUrl(sampleImageUrl, sampleStatus)
+    val repository: DogImageRepository = mockk {
+        every { getDogImageUrl() } returns Single.just(sampleDog)
+        every { insertIntoLocalDatabase(any()) } returns Completable.complete()
+    }
+    val schedulersForTests = TestSchedulersProvider()
+
+    @Test
+    fun `when viewmodel finish init, livedata should contain dog imiage url`() {
+        val observer = Observer<String> { }
+
+        val tested = DogImageViewModel(repository, schedulersForTests)
+        tested.dog.observeForever(observer)
+        val imageUrl = tested.dog.value ?: ""
```

Comment:
You can name this val `result` to indicate that it is the result of this test. And what is the reason to use `?: ""` here? Why you want to have empty string in case of null?

Reply:
because 'should be equal' method doesn't take nullables.

Reply:
Strange. I see that you are using an outdated version of Kluent, use the newest version (https://mvnrepository.com/artifact/org.amshove.kluent/kluent-android/1.68) and I think you should be able to use nullable in assertions. Also, you only need `kluent-android` dependency, `kleunt` can be removed.